### PR TITLE
VACMS-13863: Improve workflow for promo blocks

### LINKED ISSUE #13863: BE: Improve News & Benefit promo blocks workflow and help text
### User story

**AS A** Homepage Manager role in Drupal
**I WANT** clear workflow and help text when editing custom blocks (News promo, Benefit promo)
**SO THAT** I ensure the homepage works well for Veterans. 

### Notes / background

[Figma for News promo block UI](https://www.figma.com/file/sOMpovTM82TgC08ugF0FxF/News-Promo-Block?type=design&node-id=0-1&mode=design&t=4HX5hbqW1RqCYoA5-0)
- updates are pointed out in comments
- comments will also indicate which changes also apply to Benefit promo block UI

Dave C edited the News promo block and made the following observations:
- We need to do some QA on the help text associated with any of these fields.
- There's no way to "cancel" my edits when I've started editing.
- Revision logs are not required
- It's not clear why the Link Text CTA is a dropdown if the only option is "Read the full article". (Reason: #11172 will add further options.)

**Where to find in CMS:**
Filter block type on [Staging blocks list](https://staging.cms.va.gov/admin/structure/block/block-content)

[Overview of Homepage Editor UX](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/11802) of steps and UI (possibly slightly outdated)

### New AC from Tues, 8/15
**AS A** Homepage Manager (HPM)
**I WANT** to navigate to and edit News and Benefit promo blocks
**SO THAT** I can do my job.
- [x] HPM can
  - [x] Navigate to News and Benefits promo blocks using menus (Structure)

### Existing ACs from planning:
- [x] HPM only has two actions available on News and Benefit promo blocks: "Save draft" and "Publish" (both are available in the `Change to:` dropdown menu)
- [x] News Promo help text
  - [x] Block description: `This is used for identifying the block within the CMS. The description will not be shown on VA.gov.`
  - [x] Link-related text:  Link Text field on the News Promo. Text: `This text tells Veterans what to do next.`
  - [x] "Promo headline" field: make help text, `Use this headline to make it clear to Veterans what the topic of this promo is and why it's relevant to them.`
- [x] Benefit Promo help text
  - [x] Block description: `This is used for identifying the block within the CMS. The description will not be shown on VA.gov.`
  - [x] "Promo headline" field: make help text, `Use this headline to make it clear to Veterans what the topic of this promo is and why it's relevant to them.`

### ACs that need to move to new ticket due to unforeseen complexity re workflows: #14800
- [ ] Revision log required to save News promo block
- [ ] Revision log required to save Benefit promo block

### PR DESCRIPTION
**Pull request by chri5tia**
## Description

Closes #13863 

Updates text and workflow for promo blocks (news, benefit) according to [this figma mockup](https://www.figma.com/file/sOMpovTM82TgC08ugF0FxF/News-Promo-Block?type=design&node-id=0-1&mode=design&t=4HX5hbqW1RqCYoA5-0) and these concepts:

~~- Makes sure "Delete" option doesn't show up for non-admin users~~
~~- Ensures that user role "homepage manager" can't archive block~~
- Ensures that non-admins can archive block
- Updates news promo block help text: block description, link text, promo headline field
- Updates benefit promo block help text: block description, promo headline field

## Testing done
Functional testing for the promo blocks, non-promo blocks, nodes in the bundle and not in the bundle.

## Screenshots
TBD

## QA steps

As user `homepagemanager` with homepage manager role (or grant a user both homepage manager role and section `-- Office of Public and Intergovernmental Affairs` access:
- Go to [homepage news promo block edit form](https://cms-hwzr9sq9w6ph4nma2iqpmdgctt7odwpo.ci.cms.va.gov/block/209?destination=/admin/structure/block/block-content) directly; not in menu
- [x] Ensure that section settings "change to" options shows Draft and Publish and not Archive options.
- [x] Ensure that the save button at the bottom does not have "Delete" option next to it.
~~- [ ] Fill out test data in all required fields and click "Save". Ensure that there are no error messages.~~

Check the help text:
News Promo help text
- [x] Block description: _This is used for identifying the block within the CMS. The description will not be shown on VA.gov._
- [x] Link-related text: Link Text field on the News Promo. Text: _This text tells Veterans what to do next._
- [x] "Promo headline" field: _Use this headline to make it clear to Veterans what the topic of this promo is and why it's relevant to them._
 
Benefit Promo help text
- [x] Block description: _This is used for identifying the block within the CMS. The description will not be shown on VA.gov._
- [x] "Promo headline" field: _Use this headline to make it clear to Veterans what the topic of this promo is and why it's relevant to them._

### Definition of Done

- [ ] Documentation has been updated, if applicable.
- [ ] Tests have been added if necessary.
- [ ] Automated tests have passed.
- [ ] Code Quality Tests have passed.
- [ ] Acceptance Criteria in related issue are met.
- [ ] Manual Code Review Approved.

### Select Team for PR review

- [ ] `CMS Team`
- [x] `Public websites`
- [ ] `Facilities`
- [ ] `User support`
- [ ] `Accelerated Publishing`

### Is this PR blocked by another PR?

- [ ] `DO NOT MERGE`

### Does this PR need review from a Product Owner

- [ ] `Needs PO review`

### CMS user-facing announcement

Is an announcement needed to let editors know of this change?
- [ ] Yes, and it's written in issue ____ and queued for publication.
  - [ ] Merge and ping the UX writer so they are ready to publish after deployment
- [ ] Yes, but it hasn't yet been written
  - [ ] Don't merge yet -- ping the UX writer to write and queue content
- [ ] No announcement is needed for this code change.
  - [ ] Merge & carry on unburdened by announcements

### CODE REVIEW

**Comment by chri5tia** [comment]
After talking with Wes, we need to define the role of Homepage Editor, which currently has no users. There are some permissions there so maybe that has already been done but we would need to 
- add test users to that roll in order to test it
- make sure the permissions are what they should be

**Comment by chri5tia** [comment]
Note: Be sure that whatever user/test user is given "homepage manager role", that they are also added to the `-- Office of Public and Intergovernmental Affairs` section in order to see the blocks. And they will show up in the menu under structure > entity queue

File: config/sync/user.role.homepage_manager.yml [hunk 1]
```diff
@@ -19,6 +19,8 @@ label: 'Homepage Manager'
 weight: 5
 is_admin: null
 permissions:
+  - 'access administration pages'
+  - 'administer menu'
```
Comment (chri5tia) [inline config/sync/user.role.homepage_manager.yml:23]: Allows homepage manager to find the entityqueue in the menu

File: config/sync/user.role.homepage_manager.yml [hunk 2]
```diff
@@ -36,7 +38,6 @@ permissions:
   - 'update home_page_news_spotlight entityqueue'
   - 'update v2_home_page_create_account entityqueue'
   - 'use editorial transition approve'
-  - 'use editorial transition archive'
```
Comment (chri5tia) [inline config/sync/user.role.homepage_manager.yml:39]: Disallows homepage manager to archive the promo blocks.

**Review by dsasser — COMMENTED** [review]
Code looks good, but I've found that Beth Potts (not an admin) is still seeing the 'Archive' option on https://cms-hwzr9sq9w6ph4nma2iqpmdgctt7odwpo.ci.cms.va.gov/block/208?destination=/admin/structure/block/block-content%3Finfo%3D%26type%3Dbenefit_promo.

Screenshot 2023-08-29 at 11 03 51 AM

**Review by dsasser — APPROVED** [review]
Confirmed in TB
Screenshot 2023-08-29 at 2 58 44 PM

**Pull request merged**